runtime/gc_base/StringTable.cpp

```diff
@@ -556,9 +594,19 @@ j9gc_createJavaLangString(J9VMThread *vmThread, U_8 *data, UDATA length, UDATA s
 					break;
 				}
 			}
+			isASCIIorLatin1 = isASCII;
+			if (!isASCII && J2SE_VERSION(vm) >= J2SE_V17) {
+				for (UDATA i = 0; i < length; ++i) {
```

Comment:
This shouldn't loop over the data twice. Above we check for characters > 0x7F, this can be extended to also check for > 0xFF.

Reply:
Another way to look at it, this loop can start at the first non-ascii character as found in the check above, it doesn't need to start again from zero. Similarly below.

Reply:
I didn't do this way to avoid introducing extra overhead for pre-JDK17 levels. I am fine with either way.

Reply:
> Another way to look at it, this loop can start at the first non-ascii character as found in the check above, it doesn't need to start again from zero. Similarly below.

Agreed.

Reply:
Or compute the max char value, exiting the loop if it exceeds 0xFF.
If that max is not more than 0x7F, it is ASCII; if no more than 0xFF it is Latin1.

Reply:
Technically that introduces an overhead for pre-JDK17 levels which initially might exit the loop as early as first byte check, or even for JDK17+ as well.

Reply:
The upper bound (0xFF or 0x7F) can be set based on JDK version.

Reply:
Moved `isASCIIorLatin1` computation into `isASCII` loop.